packages/inference/src/providers/wavespeed-ai.ts

```diff
@@ -0,0 +1,178 @@
+import { InferenceOutputError } from "../lib/InferenceOutputError";
+import type { ImageToImageArgs } from "../tasks";
+import type { BodyParams, HeaderParams, RequestArgs, UrlParams } from "../types";
+import { delay } from "../utils/delay";
+import { omit } from "../utils/omit";
+import { base64FromBytes } from "../utils/base64FromBytes";
+import {
+	TaskProviderHelper,
+	TextToImageTaskHelper,
+	TextToVideoTaskHelper,
+	ImageToImageTaskHelper,
+} from "./providerHelper";
+
+const WAVESPEEDAI_API_BASE_URL = "https://api.wavespeed.ai";
+
+/**
+ * Common response structure for all WaveSpeed AI API responses
+ */
+interface WaveSpeedAICommonResponse<T> {
+	code: number;
+	message: string;
+	data: T;
+}
+
+/**
+ * Response structure for task status and results
+ */
+interface WaveSpeedAITaskResponse {
+	id: string;
+	model: string;
+	outputs: string[];
+	urls: {
+		get: string;
+	};
+	has_nsfw_contents: boolean[];
+	status: "created" | "processing" | "completed" | "failed";
+	created_at: string;
+	error: string;
+	executionTime: number;
+	timings: {
+		inference: number;
+	};
+}
+
+/**
+ * Response structure for initial task submission
+ */
+interface WaveSpeedAISubmitResponse {
+	id: string;
+	urls: {
+		get: string;
+	};
+}
+
+type WaveSpeedAIResponse<T = WaveSpeedAITaskResponse> = WaveSpeedAICommonResponse<T>;
+
+abstract class WavespeedAITask extends TaskProviderHelper {
+	private accessToken: string | undefined;
+
+	constructor(url?: string) {
+		super("wavespeed-ai", url || WAVESPEEDAI_API_BASE_URL);
+	}
+
+	makeRoute(params: UrlParams): string {
+		return `/api/v2/${params.model}`;
+	}
+	preparePayload(params: BodyParams): Record<string, unknown> {
+		const payload: Record<string, unknown> = {
+			...omit(params.args, ["inputs", "parameters"]),
+			...(params.args.parameters as Record<string, unknown>),
+			prompt: params.args.inputs,
+		};
+		// Add LoRA support if adapter is specified in the mapping
```

Comment:
We don't need to cast into `Result<string, unknown>` if the `params` have the proper type
`ImageToImageArgs`, `TextToImageArgs`, and `TextToVideoArgs` need to be improrted from `"../tasks"`
```suggestion
	preparePayload(params: BodyParams<ImageToImageArgs | TextToImageArgs | TextToVideoArgs>): Record<string, unknown> {
		const payload: Record<string, unknown> = {
			...omit(params.args, ["inputs", "parameters"]),
			...params.args.parameters,
			prompt: params.args.inputs,
		};
		// Add LoRA support if adapter is specified in the mapping
```

Reply:
It has been modified as suggested